### PR TITLE
Calculator: [Phase 4] Deprecate Perl Backend

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

Deprecation of the perl backend. All of the heavy lifting should now be done with JS.

**Remaining Todo**

- [ ] Handle overtriggering
- [x] Rewrite query based on searched expression
- [ ] <strike>Show rewritten query instead of raw query</strike>
- [x] Handle errors from initial search

## Related Issues and Discussions

Fixes: #3942
Fixes: #3267 
Fixes: #3935

## People to notify

@moollaza 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [x] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [x] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/calculator

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Calculator.pm [hunk 1]
```diff
@@ -108,131 +129,46 @@ handle query_nowhitespace => sub {
...
         $query =~ s#\b$name\b#($name)#ig;
     }
     my @numbers = grep { $_ =~ /^$number_re$/ } (split /\s+/, $tmp_expr);
     my $style = number_style_for(@numbers);
     return unless $style;
 
-    $tmp_expr = $style->for_computation($tmp_expr);
-    $tmp_expr =~ s/Math::BigInt->new\(([^)]+)\)/Math::BigInt->new\($1\)->bfac->bstr/g;    #correct expression for fact
     # Using functions makes us want answers with more precision than our inputs indicate.
     my $precision = ($query =~ $funcy) ? undef : ($query =~ /^\$/) ? 2 : max(map { $style->precision_of($_) } @numbers);
```
Comment (moollaza) [inline lib/DDG/Goodie/Calculator.pm]: We can remove this 👍

Comment (moollaza) [inline lib/DDG/Goodie/Calculator.pm]: Also the line where `$funcy` is defined 👍

File: share/goodie/calculator/calculator.js [hunk 2]
```diff
@@ -86,48 +87,65 @@ DDH.calculator = DDH.calculator || {};
...
         var expression = expression
 
-            // 1. handles +/- percentages
-            .replace(/(\+) (\d+(\.\d{1,})?)%/g, PercentageNormalizer.addPercentage)
-            .replace(/(\d+(\.\d{1,})?) - (\d+(\.\d{1,})?)%/g, PercentageNormalizer.subtractPercentage)
-            .replace(/(\d+(\.\d{1,})?)%/g, PercentageNormalizer.soloPercentage)
-
-            // 2. handles basic arithmetic
+            // 1. handles natural language
+            .replace(/math/gi, '')
```
Comment (moollaza) [inline share/goodie/calculator/calculator.js]: Perhaps we should only do these replacements when we take the query from the data only once.

We don't need to run these after we evaluate the query that comes from the Perl right?

**Comment by moollaza** [comment]
@pjhampton Thanks! This LGTM 👍

**Comment by bsstoner** [comment]
This was released to production 🚀

**Pull request merged**